examples/clone/clone.bal

```diff
@@ -36,34 +35,31 @@ public function main() returns (Person, Person, string)? {
         address: address
     };
 
-    // Assign the `person` value to `anyValue`. So the type of the value has to be determined in runtime.
+    // Assign the `person` value to an `any` typed variable `anyValue`.
+    // The type of the value held by `anyValue` will be determined at run time.
     any anyValue = person;
 
-    // Invoke the clone builtin method. The return type is either `Person` or `error`. Note that the `error` is returned
-    // if the value being cloned is not of `anydata` type. If the type of the value being cloned can be determined in
-    // compile time, return value is exactly of the type of the value being cloned.
-    // e.g. Person result = person.clone();
-    // In above example, 'person' is cloned instead of 'anyValue'. In compile time itself, we know the return type
-    // should be 'Person'
+    // Invoke the `.clone()` built-in method. The return type is either `Person` or `error`. Note that `error` could be 
+    // returned if the variable on which `.clone()` is called is not of type `anydata`.
+    // If the type of the value being cloned can be determined at compile time and is `anydata`, the type of the
+    // return value is exactly the type of the value being cloned.
+    // e.g. `Person result = person.clone();`
+    // If `person` is cloned instead of `anyValue`, it would be known at compile time that the return type should be
+    // `Person`.
     var result = anyValue.clone();
 
-    // Type of any variable is only determined in runtime. Therefore we have to check if the returned type is `Person`
-    // or an `error`. If we cloned the `person` value, we do not need to check the type of the return value.
+    // The type of the value held by an `any` typed variable is only determined at run time.
+    // Therefore the type of the returned value on a clone attempt could be `error`, in case a non-`anydata` value is
+    // found.
+    // Check if the `.clone()` attempt is successful, and thus returns a `Person` value.
     if (result is Person) {
         io:println("Source value: ", person);
         io:println("Cloned value: ", result);
-        string refCheck = "";
-        // Check if the reference of both values is not the same.
-        if (result !== person) {
-            refCheck = "Source and Clone are at two different memory locations";
-            io:println(refCheck);
-        }
-        return (person, result, refCheck);
-
-     // If the result is an error, the error is printed out and () is returned.
-    } else {
-        io:println("Cannot clone: ", result.reason());
+        // Check reference inequality for the original value and the cloned value.
+        io:println("Source and Clone are at two different memory locations: ", result !== person);
+    // If the result is an error, print out the detail message.
```

Comment:
```suggestion
    // If the result is an error, print out the `detail()` message.
```

Apply elsewhere if applicable.

Reply:
`detail()` returns a map, what we are doing is retrieving a message from the map. This depends on the implementation too. Not all errors would have a `message` entry in the detail map. So shall we leaves as is? WDYT?

Reply:
If it's not a technical term, the language doesn't read correctly here. That's my concern. If it isn't within back ticks it should probably read as "print out the detailed message."

Reply:
Will change to 
```suggestion
    // If the result is an error, print out the detailed error message.
```